projects/igniteui-angular/src/lib/grids/grid/grid-validation.service.ts

```diff
@@ -137,6 +137,22 @@ export class IgxGridValidationService {
         this.updateStatus();
     }
 
+    /**
+     * @hidden
+     * @internal
+     * Update validity based on new data.
+     */
+    public updateAll(newData: any) {
+        if (!newData) return;
```

Comment:
You can also add another short-circuit here to avoid looping the data -  if the `_validityStates` map is empty